CMakeLists.txt

```diff
@@ -28,6 +28,11 @@ elseif(NOT CMAKE_BUILD_TYPE IN_LIST ALLOWABLE_BUILD_TYPES)
     choose one from ${ALLOWABLE_BUILD_TYPES}")
 endif()
 
+string(TOUPPER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE_UPPERCASE)
+if("${CMAKE_BUILD_TYPE_UPPERCASE}" STREQUAL "DEBUG")
```

Comment:
Perhaps you could comment that we added these so that cuda-memcheck would give more useful
stack traces?  We don't want to think it was required, later on.

Reply:
Thanks, comments are added.